float64[3] left_xyz
float64[3] right_xyz
int num
---
bool success
